beamer/tests/agent/test_challenge.py

```diff
@@ -491,9 +571,99 @@ def test_invalidation(
         while agent.get_context(direction).l1_resolutions.get(claim.invalidation_tx) is None:
             sleeper.sleep(0.1)
 
+    ape.chain.provider._make_request(
+        "evm_setAccountBalance",
+        [l1_messenger.address, encode_hex(str(ape.convert("1 ether", int)))],
+    )
+
     request_manager.invalidateFill(request_id, FILL_ID, ape.chain.chain_id, sender=l1_messenger)
 
     ape.chain.mine(timestamp=claim.termination + 100)
     with Sleeper(5) as sleeper:
         while not claim.invalidated_l1_resolved.is_active:
             sleeper.sleep(0.1)
+
+
+@pytest.mark.usefixtures("setup_relayer_executable")
+def test_avoid_draining_agent_funds(
+    request_manager, token, agent, direction, chain_params, finality_period, l1_messenger
+):
+    requester, target, exploiter = alloc_accounts(3)
+    ape.accounts.test_accounts[0].transfer(exploiter, ape.convert("10 ether", int))
+    amount = 1
+
+    new_chain_params = (1, *chain_params[1:])
+    request_manager.updateChain(ape.chain.chain_id, *new_chain_params)
+
+    request_id = make_request(request_manager, token, requester, target, amount)
+
+    with Sleeper(5) as sleeper:
+        while (request := agent.get_context(direction).requests.get(request_id)) is None:
+            sleeper.sleep(0.1)
+
+    with Sleeper(5) as sleeper:
```

Comment:
this check is not necessary as we wait for the claim in the lines below. The claim will only come if the request was seen and claimed